remote/remote_test.go

```diff
@@ -1496,3 +1510,61 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 		})
 	})
 }
+
+func TestRemoteRetryLogic(t *testing.T) {
+	testCases := []RemoteTestCase{
+		{name: "Do not retry after status code 200 ", repo: "org/do-not-retry", statusCode: http.StatusOK, failedCount: 0, expected: 1},
+		{name: "Retry after status code 404", repo: "org/retry-not-found", statusCode: http.StatusNotFound, failedCount: 2, expected: 3},
+		{name: "Retry after status code 401", repo: "org/retry-unauthorized", statusCode: http.StatusUnauthorized, failedCount: 2, expected: 3},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			img, _ := random.Image(1024, 1)
+			configName, _ := img.ConfigName()
+			server := createServer(img, configName, tc)
+			u, err := url.Parse(server.URL)
+			if err != nil {
+				t.Fatalf("url.Parse(%v) = %v", server.URL, err)
+			}
+			defer server.Close()
+			var repoName = u.Hostname() + ":" + u.Port() + "/" + tc.repo
+
+			// exercise the subject
+			count = 0
+			_, err = remote.NewImage(repoName, authn.DefaultKeychain, remote.WithPreviousImage(repoName))
+			if err != nil {
+				t.Fatalf("Error: %v, running remoteNewImage", err)
+			}
```

Comment:
```suggestion
			h.AssertNil(t, err)
```

Reply:
For consistency with existing patterns in this file. You could argue however that the way you have it produces a more descriptive error message...

Reply:
Applied